docs/datagen/loottables.md

```diff
@@ -84,7 +83,7 @@ If `DeferredRegister` is being used to register a mod's objects, then the `#getK
 protected Iterable<Block> getKnownBlocks() {
   return BLOCK_REGISTRAR.getEntries() // Get all registered entries
     .stream() // Stream the wrapped objects
-    .flatMap(RegistryObject::stream) // Get the object if available
+    .flatMap(holder -> holder.asOptional().stream()) // Get the object if available
     ::iterator; // Create the iterable
```

Comment:
Creating an `Iterable` from a `Stream` this way is a very bad idea (especially in documention intended for beginners) because an `Iterable` is expected to be able to provide a fresh `Iterator` multiple times while a `Stream` can only do so once as it only allows a single terminal operation to be done on it.

This would be the better solution to provide all registered blocks:
```java
return BLOCK_REGISTRAR.getEntries()
    .stream()
    .map(Holder::value)
    .toList();
```